primitives/externalities/src/lib.rs

```diff
@@ -173,6 +173,11 @@ pub trait Externalities: ExtensionStore {
 	/// Returns the SCALE encoded hash.
 	fn storage_changes_root(&mut self, parent: &[u8]) -> Result<Option<Vec<u8>>, ()>;
 
+	/// Return handle that can spawn parallel tasks.
+	///
+	/// Can be none if host does not support parallel tasks.
+	fn spawn_handle(&self) -> Option<&dyn ClonableSpawn>;
```

Comment:
I really don't want to add anymore functionality to `Externalities`. This was the reason `Extension`'s where introduced. I know that extension handling is currently a bit shitty, but please make this an extension.

Reply:
Fine with me, but just to note and as discussed: this `Externalities` ability will be quite essential (modulo test implementations of it), while `Extension`-s are optional and generally not required for code execution.

Reply:
Ok after some more disussion with @kianenigma there is a strong indication that this functionality will need to be rather configurable, and Extension will definitely help to achieve configurability much in much more convenient way.

Reply:
Mot notably, if we can configure the child threads (or whatever we call them in this context) to not be able to write to storage and instead return the buffer changes, this will give us a big chunk of what we need for transaction parallelism.

Reply:
@bkchr updated!